lib/sklansky.dart

```diff
@@ -66,22 +65,100 @@ class Sklansky extends ParallelPrefix {
   }
 }
 
+class KoggeStone extends ParallelPrefix {
+  KoggeStone(List<Logic> inps, op) : super(inps, 'kogge_stone') {
+    final List<Logic> iseq = [];
+
+    inps.forEachIndexed((i, el) {
+      iseq.add(addInput('i$i', el, width: el.width));
+      _oseq.add(addOutput('o$i', width: el.width));
+    });
+
+    var skip = 1;
+
+    while (skip < inps.length) {
+      for (var i = inps.length - 1; i >= 0; --i) {
+        if (i - skip >= 0) {
+          var tmp = Logic(width: iseq[i].width);
+          tmp <= op(iseq[i - skip], iseq[i]);
+          iseq[i] = tmp;
+        }
+      }
+      skip *= 2;
+    }
+
+    iseq.forEachIndexed((i, el) {
+      _oseq[i] <= el;
+    });
+  }
+}
+
+class BrentKung extends ParallelPrefix {
+  BrentKung(List<Logic> inps, op) : super(inps, 'brent_kung') {
+    final List<Logic> iseq = [];
+
+    inps.forEachIndexed((i, el) {
+      iseq.add(addInput('i$i', el, width: el.width));
+      _oseq.add(addOutput('o$i', width: el.width));
+    });
+
+    // Reduce phase
+    var skip = 2;
+    while (skip <= inps.length) {
+      for (var i = skip - 1; i < inps.length; i += skip) {
+        var l = i - skip ~/ 2;
+        var tmp = Logic(width: iseq[i].width);
+        tmp <= op(iseq[l], iseq[i]);
+        iseq[i] = tmp;
+      }
+      skip *= 2;
+    }
+
+    // Prefix Phase
+    skip = largestPow2LessThan(inps.length);
+    while (skip > 2) {
+      for (var i = 3 * (skip ~/ 2) - 1; i < inps.length; i += skip) {
+        var l = i - skip ~/ 2;
+        var tmp = Logic(width: iseq[i].width);
+        tmp <= op(iseq[l], iseq[i]);
```

Comment:
Why do you create the `tmp` variable instead of assigning the output of `op` directly into `iseq`?

Reply:
I want a new Logic variable---there is probably a better way to create this. I don't think this is the same thing as '<='. Can I just use '=' here?
I'm overwriting the iseq list with new variables as I implement each row of a prefix table (there should be one for each black dot.)

![image](https://user-images.githubusercontent.com/1512574/200639301-87f21e81-9979-46f6-a70c-425f2d62a061.png)

Reply:
Functionally, what you have:
```dart
var tmp = Logic(width: iseq[i].width);
tmp <= op(iseq[l], iseq[i]);
iseq[i] = tmp;
```
should be equivalent to this:
```dart
iseq[i] = op(iseq[l], iseq[i]);
```
since the `op` returns a `Logic` already.

If you want to create an additional `Logic` along the path, you can do what you did, but I'm not sure what you gain.  Even the generated SystemVerilog will most likely collapse the unnecessary `assign` statements of one signal to another into one signal.

If there's some reason I'm not understanding, why you want this still, you could also do something like this:
```dart
iseq[i] = Logic(width: iseq[i].width)
  ..gets(op(iseq[l], iseq[i]));
```